### ISSUE TITLE
Case object of sealed ADT

### ISSUE THREAD

**Issue by muradm** [vectos/formulation#1]
**Version**: (e.g. `0.4.2`)
**Module**: (e.g. `formulation-core`)

### Expected behavior
Allow ADT with object case classes, like:
```
sealed trait ADT
case object Initial extends ADT
case class State1(arg: String) extends ADT
```

### Actual behavior
Currently there is no way to define case object of sealed trait.

### Steps to reproduce the behavior
Try to define `Avro[ADT]` above.

**Comment by Fristi** [comment]
Good point, I think you could fix this by adding a `record0` combinator

**Comment by muradm** [comment]
Already done :) looking at comments for first commit.
